src/aiida/engine/daemon/execmanager.py

```diff
@@ -360,15 +361,14 @@ def _copy_local_files(logger, node, transport, inputs, local_copy_list):
             if data_node.base.repository.get_object(filename_source).file_type == FileType.DIRECTORY:
                 # If the source object is a directory, we copy its entire contents
                 data_node.base.repository.copy_tree(filepath_target, filename_source)
+                transport.put(f'{dirpath}/*', target or '.')
```

Comment:
Similar to my comment below, I was wondering if this means that the `local_copy_list` behavior is once again different from `cp`. Funny enough, it does seem similar when using `-r` _and_ adding a forward slash after the source directory:


```
❯ rm -rf *; mkdir pseudo; cp -r ../test_qe/pseudo ./pseudo; tree
.
└── pseudo
    └── pseudo
        ├── Ba.upf
        └── Si.upf

3 directories, 2 files
❯ rm -rf *; mkdir pseudo; cp -r ../test_qe/pseudo/ ./pseudo; tree
.
└── pseudo
    ├── Ba.upf
    └── Si.upf

2 directories, 2 files
```

Kind of similar to `rsync`, I guess.

Reply:
Hmm, actually, after having a closer look, the behaviour seems different than I expected. With the following code:


```python
import shutil

from logging import LoggerAdapter

from aiida import orm, load_profile
from aiida.common import AIIDA_LOGGER
from aiida.common.folders import SandboxFolder
from aiida.engine.daemon.execmanager import _copy_local_files, _copy_sandbox_files

load_profile()

random_calc_job = orm.load_node(36)
logger = LoggerAdapter(logger=AIIDA_LOGGER.getChild('execmanager'))

localhost = orm.load_computer('localhost')

remote_workdir = '/Users/mbercx/project/core/jupyter/workdir'
test_path = '/Users/mbercx/project/core/jupyter/test_qe'

shutil.rmtree(remote_workdir, ignore_errors=True)

folder_data = orm.FolderData(tree=test_path)
folder_data.store()

local_copy_list = [
    (folder_data.uuid, 'pseudo', 'pseudo'),
]

with SandboxFolder() as sandbox_folder:

    sandbox_folder.get_subfolder('pseudo', create=True)

    with localhost.get_transport() as transport:

        transport.mkdir(remote_workdir)
        transport.chdir(remote_workdir)

        _copy_sandbox_files(logger, random_calc_job, transport, sandbox_folder)
        _copy_local_files(logger, random_calc_job, transport, None, local_copy_list)
```

and the contents of `test_qe`:

```
test_qe
└── pseudo
    ├── Ba.upf
    └── Si.upf
```

I indeed get a nested folder:

```
workdir/
└── pseudo
    └── pseudo
        ├── Ba.upf
        └── Si.upf
```

Not creating the `pseudo` folder in the sandbox leads to the non-nested result. However, if I make the `local_copy_list`:

```
local_copy_list = [
    (folder_data.uuid, 'pseudo', '.'),
]
```

Then the `workdir` becomes:

```
workdir/
├── Ba.upf
└── Si.upf
```

Is that what we want? Now we are really doing the `cp -r pseudo/ pseudo` (with forward slash) option, which means copy _the contents_ of the directory to the target path.